### ISSUE TITLE
cpp: gigasecond does not compile

### ISSUE THREAD

**Issue by siebenschlaefer** [exercism/exercism#5729]
After a student told me about their problems with compiling "gigasecond" in the online editor I took my old correct solution and tried to compile it. I got the error
```
We received the following error when we ran your code:

make: *** No targets specified and no makefile found.  Stop.
```
I don't know the infrastructure well enough to make an educated guess why that error occurs.

If it helps: "gigasecond" is the only exercise that requires an external library: Boost

Let me know if I should provide some more details.

**Comment by angelikatyborska** [comment]
All external libraries required for exercises need to be installed in the Docker container (when it gets build) because it runs without network access.

If you know how to do that, you could add a command to this Dockerfile: https://github.com/exercism/cpp-test-runner/blob/main/Dockerfile#L16

**Comment by iHiD** [comment]
@exercism/cpp Any ideas on this further to what Angelika has said? Missing a makefile sounds like maybe something is missing?

**Comment by silvanocerza** [comment]
I wouldn't expect a missing library to cause that error really, though it makes sense that an exercise won't compile if an external library is not installed in the Docker image running the tests runner.

I'll investigate, no ETA for now, kinda ectic these days for me.

**Comment by peterschussheim** [comment]
Perhaps we should add tests that ensure the relevant Boost libraries are installed in the test-runner container?

---

I did a simple test consisting of adding boost to the dockerfile.  Unfortunately, I am not familiar with the way docker is used in relation to the rest of the Exercism system which is why I suspect the command below doesn't confirm if the problem here is due to a missing library in the container:

```bash
.\bin\run-in-docker.sh example_success .\tests\example-success\ .\tests\example-success\out\
```

This built the docker image successfully but the terminal window closes so quickly I cannot read the details of the script operations.

**Comment by ErikSchierboom** [comment]
@peterschussheim:

> Perhaps we should add tests that ensure the relevant Boost libraries are installed in the test-runner container?

Absolutely! Any dependencies that exercise have should be pre-installed in the test-runner container. @silvanocerza do you have a list of which dependencies are being used?

> This built the docker image successfully but the terminal window closes so quickly I cannot read the details of the script operations.

To find out whether it was successful, you can examine the `results.json` file in the output directory (`.\bin\run-in-docker.sh example_success .\tests\example-success\ .\tests\example-success\out\results.json`). It's top-level `status` property should be `pass` or `fail`, not `error`.

Could you perhaps open a PR to add boost to the Dockerfile? I can then test things locally.
